pages/discover.js

```diff
@@ -11,40 +11,43 @@ export async function getServerSideProps(context) {
     random: [],
     trending: [],
   };
+
+  const popularPromise = fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/popular`
+  );
+  const randomPromise = fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/random`
+  );
+  const tagsPromise = fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/tags`
+  );
+  const trendingPromise = fetch(
+    `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/trending`
+  );
+
   try {
-    const res = await fetch(
-      `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/popular`
-    );
-    data.popular = await res.json();
-  } catch (e) {
+    data.popular = await (await popularPromise).json()
```

Comment:
I think a `promise all` would be more efficient otherwise this one is still blocking the remaining waits in this order

Reply:
The calls have been initiated so when we `wait for one` the rest of them are either processed or running in the background

the eg:- we are waiting for the `popularPromise` but other 3 promises are running so when we finish with the `popularPromise` rest of them are either finished or about to finish so 

rest of the `await (await Promise).json()` will not taking any time or take a very less time 

Plus the `Promise.all` is all or nothing so if we end up `rejecting one promise` the whole list of promises reject hence getting an `empty data as prop`

Reply:
All requests are to the same server, it is unlikely 1 will throw an exception. If we are trying to improve the performance, we should use promise all to maximise the performance and truly run them in parallel

Reply:
ahh, Got it